### PR TITLE
source-archive: Correct version modification time and short version

### PR DESCRIPTION
**Pull request by ghost**
Correct newline stripping and timestamp

```
scripts/source-archive --release=test --target=/tmp --tag=HEAD
```

Check `cmake/GitVersion.cmake` in the tar archive.  The shortversion will now not have a trailing newline.  The version will also be the current date and time, rather than 1-1-1970.  While that shouldn't have been problematic, it was breaking the Ninja build.

### CODE REVIEW

File: scripts/source-archive [hunk 1]
```diff
@@ -186,6 +186,7 @@ if __name__ == "__main__":
         vcsdate_unix, vcsdate))
     cmakeversion = tarfile.TarInfo("%s/cmake/GitVersion.cmake" % (prefix))
     cmakeversion.size = cmakeversionbuf.len
+    cmakeversion.mtime = time.time()
```
Comment (sbesson) [inline scripts/source-archive:189]: Can you add a comment in the script itself explaining the reason why the `mtime` attribute of `TarInfo` need to be overriden?

**Comment by sbesson** [comment]
Is the file removal required in this context or could we have the cleanup done as a separate PR?

**Comment by ghost** [comment]
@sbesson Yikes, no.  That's unrelated for another card.  I've set it back to the minimal change which was intended.

**Comment by ghost** [comment]
For reference: https://github.com/ninja-build/ninja/issues/1120 this is the bug report for the Ninja behaviour.

**Comment by sbesson** [comment]
Retested with the last version of the commit. Works as expected.

**Pull request merged**